packages/browser/src/browser/__tests__/integration.test.ts

```diff
@@ -897,26 +899,56 @@ describe('retries', () => {
     expect(ajs.queue.queue.getAttempts(fruitBasketEvent)).toEqual(2)
   })
 
-  it('does not queue up events when offline if retryQueue setting is set to false', async () => {
-    const [ajs] = await AnalyticsBrowser.load({ writeKey })
+  it('does not queue events / dispatch when offline if retryQueue setting is set to false', async () => {
+    const [ajs] = await AnalyticsBrowser.load(
+      { writeKey },
+      { retryQueue: false }
+    )
 
+    const trackSpy = jest.fn().mockImplementation((ctx) => ctx)
     await ajs.queue.register(
       Context.system(),
       {
         ...testPlugin,
         ready: () => Promise.resolve(true),
-        track: (ctx) => ctx,
+        track: trackSpy,
       },
       ajs
     )
 
     // @ts-ignore ignore reassining function
     isOffline = jest.fn().mockReturnValue(true)
 
-    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    ajs.track('event')
+    await ajs.track('event')
 
-    expect(ajs.queue.queue.length).toBe(0)
+    expect(trackSpy).toBeCalledTimes(0)
+  })
+
+  it('enqueues events / dispatches if the client is currently offline and retries are *enabled* for the main event queue', async () => {
+    const [ajs] = await AnalyticsBrowser.load(
+      { writeKey },
+      { retryQueue: true }
+    )
+
+    const trackSpy = jest.fn().mockImplementation((ctx) => ctx)
+    await ajs.queue.register(
+      Context.system(),
+      {
+        ...testPlugin,
+        ready: () => Promise.resolve(true),
+        track: trackSpy,
+      },
+      ajs
+    )
+
+    // @ts-ignore ignore reassining function
+    isOffline = jest.fn().mockReturnValue(true)
```

Comment:
Is this only returning `true` once? I'm wondering how come the plugin is called if `isOffline` is always true.

Reply:
The plugin will be called unless both isOffline = true _and_ retryQueue = false. So basically, it still goes through the plugins no matter what unless a user explicitly sets retryQueue to false.

Reply:
Oh interesting, I always thought if you were offline it would just go immediately to the PersistedQueue.